#!/usr/bin/env python

PACKAGE = 'stairdetect'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("debug",                       bool_t,   0, "show/hide debug output", True)
gen.add("canny_low_th",                double_t, 0, "Canny low threshold",    40, 0, 100)
gen.add("canny_ratio",                 double_t, 0, "Canny ratio",            10,  0, 20)
gen.add("canny_kernel_size",           int_t,    0, "Canny kernel size",      1,  0, 3)
gen.add("hough_min_line_length",       double_t, 0, "Hough Tf: minimum line length",                               25, 1, 200)
gen.add("hough_max_line_gap",          double_t, 0, "Hough Tf: maximum line gap",                                  15, 1, 200)
gen.add("hough_th",                    double_t, 0, "Hough Tf: Accumulator threshold parameter",                   20, 0, 500)
gen.add("hough_rho",                   int_t,    0, "Hough Tf: Distance resolution of the accumulator in pixels.", 8,  0, 20)
gen.add("hough_theta",                 double_t, 0, "Hough Tf: Angle resolution of the accumulator in radians.",   1,  1, 10)

exit(gen.generate(PACKAGE, "stairdetect", "StairDetect"))